api/src/main/java/org/eclipse/microprofile/config/inject/ConfigProperty.java

```diff
@@ -139,12 +139,17 @@
     String name() default "";
 
     /**
-     * The default value if the configured property value does not exist. Empty string as the default value will be
-     * ignored, which is same as not setting the default value.
+     * The default value if the configured property property does not exist.
      * <p>
-     * If the target Type is not String a proper {@link org.eclipse.microprofile.config.spi.Converter} will get applied.
-     * That means that any default value string should follow the formatting rules of the registered Converters.
-     *
+     * If the target Type is not String, a proper {@link org.eclipse.microprofile.config.spi.Converter} will get
+     * applied. This value acts as a configure source with the lowest ordinal.
+     * 
+     * Empty string as the default value will be ignored, which is same as not setting the default value. That means
+     * that any default value string should follow the formatting rules of the registered Converters.
+     * 
+     * If a property has been deleted by a config source via setting an empty configure value or a value causing the
+     * corresponding converter returning {@code null} with a higher ordinal, the default value will not be used.
```

Comment:
```suggestion
     * If a property has been deleted by a config source with a higher ordinal by setting an empty configuration value or by using a value causing the used converter returning {@code null}, the default value will not be used.
```
This sentence was very confusing - I hope I got it more clear in my suggestion.

To be fully clear we maybe should also state that any value (overwritten by a higher ordinal or not) causing the converter to return null will not cause use of the default value? This isn't fully clear when we just mention the override via higher ordinal source.

Reply:
+1

Reply:
I'll probably use the word "empty" or "emptied" instead of "delete" or "deleted":

"If a property has been emptied...". This is to be consistent with the empty rules wording.